com.unity.netcode.gameobjects/Runtime/Messaging/Messages/RpcMessages.cs

```diff
@@ -66,7 +66,12 @@ public static unsafe bool Deserialize(ref FastBufferReader reader, ref NetworkCo
         public static void Handle(ref NetworkContext context, ref RpcMetadata metadata, ref FastBufferReader payload, ref __RpcParams rpcParams)
         {
             var networkManager = (NetworkManager)context.SystemOwner;
-            var networkObject = networkManager.SpawnManager.SpawnedObjects[metadata.NetworkObjectId];
+
+            var objectExists = networkManager.SpawnManager.SpawnedObjects.TryGetValue(metadata.NetworkObjectId, out var networkObject);
+            if (!objectExists)
+            {
+                throw new Exception("RPC called on an object that isn't in the spawned objects list - please make sure an object is spawned before calling RPCs.");
+            }
```

Comment:
you don't actually need an extra local variable here, also minor comment suggestion:
```suggestion
            if (!networkManager.SpawnManager.SpawnedObjects.TryGetValue(metadata.NetworkObjectId, out var networkObject))
            {
                throw new Exception($"RPC called on a {nameof(NetworkObject)} that is not in the spawned objects list - please make sure the {nameof(NetworkObject)} is spawned before calling RPCs.");
            }
```